### PR TITLE
[Backport] Add class into image builder

### PR DESCRIPTION
**Pull request by gelanivishal**
### Original PR
https://github.com/magento/magento2/pull/18388

### Description
Currently, the product image has an hardcoded class `product-image-photo` which can't be overriden when using the `$attributes` parameter of the `imageBuilder->create()` function.

### Manual testing scenarios
1. Display a product image using 
```
$block->getImage($product, $image, ['class' => 'my-custom-class'])->toHtml();
```

=> The `img` tag has `my-custom-class` as class.

2. Display a product image using 
```
$block->getImage($product, $image)->toHtml();
```

=> The `img` tag has `product-image-photo` as class.

The aim here is to allow customization without breaking the default value.

I'm open to discussion about how to deal with this!

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by sidolov** [comment]
Hi @gelanivishal , changes in original PR are completely different and can't be ported automatically. 
Please, adopt changes for corresponded version.

**Comment by gelanivishal** [comment]
@sidolov Please check now.

File: app/code/Magento/Catalog/Block/Product/ImageFactory.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
+<?php
```
Comment (sidolov) [inline app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: It's not a good idea to port the whole class and use only one method. Please, consider to provide different implementation compatible with 2.2-develop.

File: app/code/Magento/Catalog/Model/Product/Attribute/Repository.php [hunk 2]
```diff
@@ -1,319 +0,0 @@
-<?php
```
Comment (sidolov) [inline app/code/Magento/Catalog/Model/Product/Attribute/Repository.php:1]: We can't remove attribute repository according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Product/ImageFactory.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: Disagree with that. @sidolov why we need to reinvent the wheel?

Comment (orlangur) [reply app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: I'm not sure if this new class is actually used though.

File: app/code/Magento/Catalog/Test/Unit/Block/Product/ImageFactoryTest.php [hunk 3]
```diff
@@ -1,212 +0,0 @@
-<?php
```
Comment (orlangur) [inline app/code/Magento/Catalog/Test/Unit/Block/Product/ImageFactoryTest.php]: No, you cannot just remove a test.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Block/Product/ImageFactory.php (hunk 1 above)
Comment (sidolov) [reply app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: @orlangur I agree that we shouldn't implement completely new solution, but introducing code that never used not a good idea. We should port only the fragments that resolves the problem

Comment (orlangur) [reply app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: @sidolov but it requires more efforts and make codebase 2.2/2.3 less consistent. I see your point but as for me some unused code is lesser evil.

Comment (sidolov) [reply app/code/Magento/Catalog/Block/Product/ImageFactory.php:1]: @orlangur it's pretty simple, for 2.2 we need just one method from 2.3 version. We trying to introduce new code that will be marked as unused and deprecated

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.
Thanks for collaboration!

**Pull request closed without merging**